### PR TITLE
Fix nullptr dereference in SynchronizeShard.

### PR DESCRIPTION
**Pull request by danielhlarkin**

### CODE REVIEW

**Comment by danielhlarkin** [comment]
http://jenkins01.arangodb.biz:8080/job/arangodb-matrix-pr/1973/

File: arangod/Cluster/SynchronizeShard.cpp [hunk 1]
```diff
@@ -443,7 +443,7 @@ arangodb::Result SynchronizeShard::getReadLock(
       std::unordered_map<std::string, std::string>(), timeout);
 
     auto result = putres->result;
-    if (result != nullptr && result->getHttpReturnCode() == 200) {
+    if (result != nullptr || result->getHttpReturnCode() == 200) {
```
Comment (jsteemann) [inline arangod/Cluster/SynchronizeShard.cpp]: this is wrong now.

File: arangod/Cluster/SynchronizeShard.cpp [hunk 2]
```diff
@@ -387,7 +387,7 @@ arangodb::Result cancelBarrier(
 
   if (comres->status == CL_COMM_SENT) {
     auto result = comres->result;
-    if (result != nullptr && result->getHttpReturnCode() != 200 &&
+    if (result != nullptr || result->getHttpReturnCode() != 200 &&
```
Comment (jsteemann) [inline arangod/Cluster/SynchronizeShard.cpp]: this is wrong now.

File: arangod/Cluster/SynchronizeShard.cpp [hunk 3]
```diff
@@ -148,7 +148,7 @@ arangodb::Result getReadLockId(
 
   auto result = comres->result;
 
-  if (result != nullptr && result->getHttpReturnCode() == 200) {
+  if (result != nullptr || result->getHttpReturnCode() == 200) {
```
Comment (jsteemann) [inline arangod/Cluster/SynchronizeShard.cpp]: this is wrong now.

File: arangod/Cluster/SynchronizeShard.cpp (hunk 1 above)
Comment (danielhlarkin) [reply arangod/Cluster/SynchronizeShard.cpp]: Oh crap, yes, you are quite right. Bad search/replace.

**Pull request merged**